diff --git a/dist/client/ssr/index.js b/dist/client/ssr/index.js
index 2c481797593a562c7566774d9bf9820a54d2508b..1a621f63176cc8708459876130bfdb291641a58e 100644
--- a/dist/client/ssr/index.js
+++ b/dist/client/ssr/index.js
@@ -34,9 +34,9 @@ _sfc_main.setup = (props, ctx) => {
 
 
 
-const ON_NODE_CLICK = Symbol(import.meta.env.DEV ? "on-file-tree-node-click" : "");
+const ON_NODE_CLICK = Symbol(import.meta.env?.DEV ? "on-file-tree-node-click" : "");
 
-const ACTIVE_NODE_KEY = Symbol(import.meta.env.DEV ? "active-file-tree-node" : "");
+const ACTIVE_NODE_KEY = Symbol(import.meta.env?.DEV ? "active-file-tree-node" : "");
 
 
 const _sfc_main$1 =  defineComponent({
