diff --git a/dist/client/theme-default/Layout.vue b/dist/client/theme-default/Layout.vue
index 7c729e1ff751c14e4bf355d160f6c3cf0d6422ef..119e3ee899ebe3c6437e6321e2b3a87544b6694e 100644
--- a/dist/client/theme-default/Layout.vue
+++ b/dist/client/theme-default/Layout.vue
@@ -1,4 +1,5 @@
 <script setup>
+// <script setup lang="ts">
 import { computed, provide, useSlots } from "vue";
 import VPBackdrop from "./components/VPBackdrop.vue";
 import VPContent from "./components/VPContent.vue";
