diff --git a/index.d.ts b/index.d.ts
index e3cdbe87101c8b070261b402a7b49c020ca37413..68519136d0df962beca1ef84a377efa4c4d4c108 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -1,5 +1,5 @@
 interface Document {
-    startViewTransition(updateCallback: () => Promise<void> | void): ViewTransition;
+    startViewTransition(updateCallback: () => Promise<void | unknown> | void | unknown): ViewTransition;
 }

 interface ViewTransition {
