diff --git a/index.d.ts b/index.d.ts
index 25bc72acab41076de886dd83a3f76536fc659502..61aa514eca574715ccc2420b51ca181fa56d3067 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -6,6 +6,7 @@
 
 import * as CSS from "csstype";
 import * as PropTypes from "prop-types";
+import { Argument as ClassArray } from "classnames";
 
 type NativeAnimationEvent = AnimationEvent;
 type NativeClipboardEvent = ClipboardEvent;
@@ -479,6 +480,7 @@ declare namespace React {
         | ReactElement
         | string
         | number
+        | bigint
         | Iterable<ReactNode>
         | ReactPortal
         | boolean
@@ -2142,19 +2144,19 @@ declare namespace React {
      */
     interface SyntheticEvent<T = Element, E = Event> extends BaseSyntheticEvent<E, EventTarget & T, EventTarget> {}
 
-    interface ClipboardEvent<T = Element> extends SyntheticEvent<T, NativeClipboardEvent> {
+    interface ClipboardEvent<T = Element> extends SyntheticEvent<T, NativeClipboardEvent>, globalThis.ClipboardEvent {
         clipboardData: DataTransfer;
     }
 
-    interface CompositionEvent<T = Element> extends SyntheticEvent<T, NativeCompositionEvent> {
+    interface CompositionEvent<T = Element> extends SyntheticEvent<T, NativeCompositionEvent>, globalThis.CompositionEvent {
         data: string;
     }
 
-    interface DragEvent<T = Element> extends MouseEvent<T, NativeDragEvent> {
+    interface DragEvent<T = Element> extends MouseEvent<T, NativeDragEvent>, globalThis.DragEvent {
         dataTransfer: DataTransfer;
     }
 
-    interface PointerEvent<T = Element> extends MouseEvent<T, NativePointerEvent> {
+    interface PointerEvent<T = Element> extends MouseEvent<T, NativePointerEvent>, globalThis.PointerEvent {
         pointerId: number;
         pressure: number;
         tangentialPressure: number;
@@ -2167,7 +2169,7 @@ declare namespace React {
         isPrimary: boolean;
     }
 
-    interface FocusEvent<Target = Element, RelatedTarget = Element> extends SyntheticEvent<Target, NativeFocusEvent> {
+    interface FocusEvent<Target = Element, RelatedTarget = Element> extends SyntheticEvent<Target, NativeFocusEvent>, globalThis.FocusEvent {
         relatedTarget: (EventTarget & RelatedTarget) | null;
         target: EventTarget & Target;
     }
@@ -2199,7 +2201,7 @@ declare namespace React {
         | "Symbol"
         | "SymbolLock";
 
-    interface KeyboardEvent<T = Element> extends UIEvent<T, NativeKeyboardEvent> {
+    interface KeyboardEvent<T = Element> extends UIEvent<T, NativeKeyboardEvent>, globalThis.KeyboardEvent {
         altKey: boolean;
         /** @deprecated */
         charCode: number;
@@ -2224,7 +2226,7 @@ declare namespace React {
         which: number;
     }
 
-    interface MouseEvent<T = Element, E = NativeMouseEvent> extends UIEvent<T, E> {
+    interface MouseEvent<T = Element, E = NativeMouseEvent> extends UIEvent<T, E>, globalThis.MouseEvent {
         altKey: boolean;
         button: number;
         buttons: number;
@@ -2246,7 +2248,7 @@ declare namespace React {
         shiftKey: boolean;
     }
 
-    interface TouchEvent<T = Element> extends UIEvent<T, NativeTouchEvent> {
+    interface TouchEvent<T = Element> extends UIEvent<T, NativeTouchEvent>, globalThis.TouchEvent {
         altKey: boolean;
         changedTouches: TouchList;
         ctrlKey: boolean;
@@ -2260,25 +2262,25 @@ declare namespace React {
         touches: TouchList;
     }
 
-    interface UIEvent<T = Element, E = NativeUIEvent> extends SyntheticEvent<T, E> {
+    interface UIEvent<T = Element, E = NativeUIEvent> extends SyntheticEvent<T, E>, globalThis.UIEvent {
         detail: number;
         view: AbstractView;
     }
 
-    interface WheelEvent<T = Element> extends MouseEvent<T, NativeWheelEvent> {
+    interface WheelEvent<T = Element> extends MouseEvent<T, NativeWheelEvent>, globalThis.WheelEvent {
         deltaMode: number;
         deltaX: number;
         deltaY: number;
         deltaZ: number;
     }
 
-    interface AnimationEvent<T = Element> extends SyntheticEvent<T, NativeAnimationEvent> {
+    interface AnimationEvent<T = Element> extends SyntheticEvent<T, NativeAnimationEvent>, globalThis.AnimationEvent {
         animationName: string;
         elapsedTime: number;
         pseudoElement: string;
     }
 
-    interface TransitionEvent<T = Element> extends SyntheticEvent<T, NativeTransitionEvent> {
+    interface TransitionEvent<T = Element> extends SyntheticEvent<T, NativeTransitionEvent>, globalThis.TransitionEvent {
         elapsedTime: number;
         propertyName: string;
         pseudoElement: string;
@@ -2841,7 +2843,7 @@ declare namespace React {
         // Standard HTML Attributes
         accessKey?: string | undefined;
         autoFocus?: boolean | undefined;
-        className?: string | undefined;
+        className?: string | ClassArray | undefined;
         contentEditable?: Booleanish | "inherit" | "plaintext-only" | undefined;
         contextMenu?: string | undefined;
         dir?: string | undefined;
@@ -3590,7 +3592,7 @@ declare namespace React {
 
         // Attributes which also defined in HTMLAttributes
         // See comment in SVGDOMPropertyConfig.js
-        className?: string | undefined;
+        className?: string | ClassArray | undefined;
         color?: string | undefined;
         height?: number | string | undefined;
         id?: string | undefined;
@@ -4147,7 +4149,7 @@ declare namespace React {
     // https://github.com/nikeee/2048-typescript/blob/master/2048/js/touch.d.ts
     // ----------------------------------------------------------------------
 
-    interface AbstractView {
+    interface AbstractView extends Window {
         styleMedia: StyleMedia;
         document: Document;
     }
