diff --git a/index.d.ts b/index.d.ts
index 00b03984285762c090b58e76437a49cdd38ebae8..b7dfcaf23ec574e75e5cd586155b96e98555e96a 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -5,6 +5,7 @@
 /// <reference path="global.d.ts" />
 
 import * as CSS from "csstype";
+import { ClassValue } from "clsx";
 
 type NativeAnimationEvent = AnimationEvent;
 type NativeClipboardEvent = ClipboardEvent;
@@ -385,7 +386,7 @@ declare namespace React {
     }
 
     interface ReactPortal extends ReactElement {
-        children: ReactNode;
+        children?: ReactNode;
     }
 
     /**
@@ -2011,19 +2012,19 @@ declare namespace React {
      */
     interface SyntheticEvent<T = Element, E = Event> extends BaseSyntheticEvent<E, EventTarget & T, EventTarget> {}
 
-    interface ClipboardEvent<T = Element> extends SyntheticEvent<T, NativeClipboardEvent> {
+    interface ClipboardEvent<T = Element> extends SyntheticEvent<T, NativeClipboardEvent>, globalThis.ClipboardEvent {
         clipboardData: DataTransfer;
     }
 
-    interface CompositionEvent<T = Element> extends SyntheticEvent<T, NativeCompositionEvent> {
+    interface CompositionEvent<T = Element> extends SyntheticEvent<T, NativeCompositionEvent>, globalThis.CompositionEvent {
         data: string;
     }
 
-    interface DragEvent<T = Element> extends MouseEvent<T, NativeDragEvent> {
+    interface DragEvent<T = Element> extends MouseEvent<T, NativeDragEvent>, globalThis.DragEvent {
         dataTransfer: DataTransfer;
     }
 
-    interface PointerEvent<T = Element> extends MouseEvent<T, NativePointerEvent> {
+    interface PointerEvent<T = Element> extends MouseEvent<T, NativePointerEvent>, globalThis.PointerEvent {
         pointerId: number;
         pressure: number;
         tangentialPressure: number;
@@ -2036,7 +2037,7 @@ declare namespace React {
         isPrimary: boolean;
     }
 
-    interface FocusEvent<Target = Element, RelatedTarget = Element> extends SyntheticEvent<Target, NativeFocusEvent> {
+    interface FocusEvent<Target = Element, RelatedTarget = Element> extends SyntheticEvent<Target, NativeFocusEvent>, globalThis.FocusEvent {
         relatedTarget: (EventTarget & RelatedTarget) | null;
         target: EventTarget & Target;
     }
@@ -2072,7 +2073,7 @@ declare namespace React {
         | "Symbol"
         | "SymbolLock";
 
-    interface KeyboardEvent<T = Element> extends UIEvent<T, NativeKeyboardEvent> {
+    interface KeyboardEvent<T = Element> extends UIEvent<T, NativeKeyboardEvent>, globalThis.KeyboardEvent {
         altKey: boolean;
         /** @deprecated */
         charCode: number;
@@ -2097,7 +2098,7 @@ declare namespace React {
         which: number;
     }
 
-    interface MouseEvent<T = Element, E = NativeMouseEvent> extends UIEvent<T, E> {
+    interface MouseEvent<T = Element, E = NativeMouseEvent> extends UIEvent<T, E>, globalThis.MouseEvent {
         altKey: boolean;
         button: number;
         buttons: number;
@@ -2119,7 +2120,7 @@ declare namespace React {
         shiftKey: boolean;
     }
 
-    interface TouchEvent<T = Element> extends UIEvent<T, NativeTouchEvent> {
+    interface TouchEvent<T = Element> extends UIEvent<T, NativeTouchEvent>, globalThis.TouchEvent {
         altKey: boolean;
         changedTouches: TouchList;
         ctrlKey: boolean;
@@ -2133,30 +2134,30 @@ declare namespace React {
         touches: TouchList;
     }
 
-    interface UIEvent<T = Element, E = NativeUIEvent> extends SyntheticEvent<T, E> {
+    interface UIEvent<T = Element, E = NativeUIEvent> extends SyntheticEvent<T, E>, globalThis.UIEvent {
         detail: number;
         view: AbstractView;
     }
 
-    interface WheelEvent<T = Element> extends MouseEvent<T, NativeWheelEvent> {
+    interface WheelEvent<T = Element> extends MouseEvent<T, NativeWheelEvent>, globalThis.WheelEvent {
         deltaMode: number;
         deltaX: number;
         deltaY: number;
         deltaZ: number;
     }
 
-    interface AnimationEvent<T = Element> extends SyntheticEvent<T, NativeAnimationEvent> {
+    interface AnimationEvent<T = Element> extends SyntheticEvent<T, NativeAnimationEvent>, globalThis.AnimationEvent {
         animationName: string;
         elapsedTime: number;
         pseudoElement: string;
     }
 
-    interface ToggleEvent<T = Element> extends SyntheticEvent<T, NativeToggleEvent> {
+    interface ToggleEvent<T = Element> extends SyntheticEvent<T, NativeToggleEvent>, globalThis.ToggleEvent {
         oldState: "closed" | "open";
         newState: "closed" | "open";
     }
 
-    interface TransitionEvent<T = Element> extends SyntheticEvent<T, NativeTransitionEvent> {
+    interface TransitionEvent<T = Element> extends SyntheticEvent<T, NativeTransitionEvent>, globalThis.TransitionEvent {
         elapsedTime: number;
         propertyName: string;
         pseudoElement: string;
@@ -2732,7 +2733,7 @@ declare namespace React {
         accessKey?: string | undefined;
         autoCapitalize?: "off" | "none" | "on" | "sentences" | "words" | "characters" | undefined | (string & {});
         autoFocus?: boolean | undefined;
-        className?: string | undefined;
+        className?: string | ClassValue | undefined;
         contentEditable?: Booleanish | "inherit" | "plaintext-only" | undefined;
         contextMenu?: string | undefined;
         dir?: string | undefined;
@@ -3530,7 +3531,7 @@ declare namespace React {
 
         // Attributes which also defined in HTMLAttributes
         // See comment in SVGDOMPropertyConfig.js
-        className?: string | undefined;
+        className?: string | ClassValue | undefined;
         color?: string | undefined;
         height?: number | string | undefined;
         id?: string | undefined;
@@ -4026,7 +4027,7 @@ declare namespace React {
     // https://github.com/nikeee/2048-typescript/blob/master/2048/js/touch.d.ts
     // ----------------------------------------------------------------------
 
-    interface AbstractView {
+    interface AbstractView extends Window {
         styleMedia: StyleMedia;
         document: Document;
     }
