diff --git a/lib/utils/detectors.js b/lib/utils/detectors.js
index 3ad71f3ff1135fb035ea2f4e72f8af1c5ed28f83..e5c7d3ab101c513a72612fe2a00f255e0c735cdd 100644
--- a/lib/utils/detectors.js
+++ b/lib/utils/detectors.js
@@ -14,6 +14,7 @@ const isValidTopLevelImport = (x, state) => {
 exports.isValidTopLevelImport = isValidTopLevelImport;
 const localNameCache = {};
 const importLocalName = (name, state, options = {}) => {
+  return name === 'default' ? 'styled' : name;
   const {
     cacheIdentifier,
     bypassCache = false
diff --git a/lib/utils/getName.js b/lib/utils/getName.js
index 90d17cc65080b7909ac051734652dc28887c04ec..bae47985316462a22c45ded34365bcd8c78a8c73 100644
--- a/lib/utils/getName.js
+++ b/lib/utils/getName.js
@@ -46,6 +46,7 @@ var _default = t => path => {
   }
 
   // identifiers are the only thing we can reliably get a name from
-  return t.isIdentifier(namedNode) ? namedNode.name : undefined;
+  const name = t.isIdentifier(namedNode) ? namedNode.name : undefined;
+  return name.replaceAll(/^Styled(?=[A-Z0-9])/g, "").replaceAll(/(?<!^)([A-Z])/g, "-$1").toLowerCase();
 };
 exports.default = _default;
