diff --git a/index.d.ts b/index.d.ts
index 09f1dde2473f93eea62128c475cf715286ddbf2c..9e76010660ddfd2dbdd77b35100d8bb8753b20f6 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -66,12 +66,20 @@ export type AnimationItem = {
     autoplay: boolean;
     loop: boolean | number;
     renderer: any;
+    animationData: any;
     animationID: string;
     assetsPath: string;
     timeCompleted: number;
     segmentPos: number;
     isSubframeEnabled: boolean;
     segments: AnimationSegment | AnimationSegment[];
+    markers: {
+        time: number;
+        duration: number;
+        payload: {
+            name: string;
+        };
+    }[];
     play(name?: string): void;
     stop(name?: string): void;
     togglePause(name?: string): void;
