diff --git a/node_modules/lottie-web/index.d.ts b/node_modules/lottie-web/index.d.ts
index a018fbc..b8d735f 100644
--- a/node_modules/lottie-web/index.d.ts
+++ b/node_modules/lottie-web/index.d.ts
@@ -66,12 +66,20 @@ export type AnimationItem = {
     autoplay: boolean;
     loop: boolean | number;
     renderer: any;
+    animationData: any;
     animationID: string;
     assetsPath: string;
     timeCompleted: number;
     segmentPos: number;
     isSubframeEnabled: boolean;
     segments: AnimationSegment | AnimationSegment[];
+    markers: {
+        time: number;
+        duration: number;
+        payload: {
+            name: string;
+        };
+    }[];
     play(name?: string): void;
     stop(name?: string): void;
     togglePause(name?: string): void;
