diff --git a/dist/parser.js b/dist/parser.js
index e36af64dbd3380f7c2ac46067755c75a18a0a242..9198402a571c91a8ffb6c47b5e2bcd7823d9a908 100644
--- a/dist/parser.js
+++ b/dist/parser.js
@@ -128,7 +128,7 @@ var Parser = /*#__PURE__*/function () {
     var _this = this;
     return function (message, errorOptions) {
       if (typeof _this.rule === 'string') {
-        return new Error(message);
+        return;
       }
       return _this.rule.error(message, errorOptions);
     };
