diff --git a/lib/tokenizer.js b/lib/tokenizer.js
index 92da2efb07cb30f112c2838136e7af8071609311..cc53e0c0d73648795d41cc0080784becc6b89863 100644
--- a/lib/tokenizer.js
+++ b/lib/tokenizer.js
@@ -317,6 +317,10 @@ function tokenizer(input, options = {}) {
 
 					currentToken = ['comment', css.slice(pos, next + 1), pos, next];
 					pos = next;
+				} else if (code === SLASH && css.charCodeAt(pos + 1) === SLASH) {
+					next = css.indexOf('\n', pos + 2);
+					currentToken = ['comment', css.slice(pos, next + 1) + " */", pos, next];
+					pos = next;
 				} else {
 					RE_WORD_END.lastIndex = pos + 1;
 					RE_WORD_END.test(css);
