diff --git a/node_modules/react-flip-move/typings/react-flip-move.d.ts b/node_modules/react-flip-move/typings/react-flip-move.d.ts
index cd80c41..b95a1b0 100644
--- a/node_modules/react-flip-move/typings/react-flip-move.d.ts
+++ b/node_modules/react-flip-move/typings/react-flip-move.d.ts
@@ -6,7 +6,6 @@
 
 import { Component, ReactElement } from 'react';
 
-export as namespace FlipMove;
 export default FlipMove;
 
 declare class FlipMove extends Component<FlipMove.FlipMoveProps, any> { }
