diff --git a/esm/vanilla/utils.mjs b/esm/vanilla/utils.mjs
index 9839d4e5130922655cae09a8f1a4a30959b562a2..b8a544d873a8a6e6640025f778d1efafd358d6e6 100644
--- a/esm/vanilla/utils.mjs
+++ b/esm/vanilla/utils.mjs
@@ -290,6 +290,7 @@ function proxyMap(entries) {
     [Symbol.iterator]() {
       return this.entries();
     },
+    __proto__: Map.prototype,
     get [Symbol.toStringTag]() {
       return "Map";
     },
@@ -463,6 +464,7 @@ function proxySet(initialValues) {
         yield [value, value];
       }
     },
+    __proto__: Set.prototype,
     toJSON() {
       return new Set(this.values());
     },
diff --git a/vanilla/utils.js b/vanilla/utils.js
index 49236b2d81a1a23b84217e556706604eb29cc30f..4e670655786c0ab4dd4aae50025c269e9a8194fb 100644
--- a/vanilla/utils.js
+++ b/vanilla/utils.js
@@ -295,6 +295,7 @@ function proxyMap(entries) {
     get [Symbol.toStringTag]() {
       return "Map";
     },
+    __proto__: Map.prototype,
     toJSON() {
       return new Map(this.entries());
     }
@@ -465,6 +466,7 @@ function proxySet(initialValues) {
         yield [value, value];
       }
     },
+    __proto__: Set.prototype,
     toJSON() {
       return new Set(this.values());
     },
