diff --git a/client.d.ts b/client.d.ts
index 0f2c3db286d82fa5f2912046cae9ce15d19b0514..82dc540d0d704ad874ba2da55597341d7dcb538b 100644
--- a/client.d.ts
+++ b/client.d.ts
@@ -130,9 +130,9 @@ declare module '*.gif' {
   const src: string
   export default src
 }
-declare module '*.svg' {
-  const src: string
-  export default src
+declare module '*.svg' { // Using with vite-plugin-svgr
+  const SvgElement: typeof import('*.svg?react')['default']
+  export default SvgElement
 }
 declare module '*.ico' {
   const src: string
@@ -147,6 +147,16 @@ declare module '*.avif' {
   export default src
 }
 
+// cursors
+declare module '*.cur' {
+  const src: string
+  export default src
+}
+declare module '*.ani' {
+  const src: string
+  export default src
+}
+
 // media
 declare module '*.mp4' {
   const src: string
