diff --git a/client.d.ts b/client.d.ts
index 7d81044009215ef258a07dcad470e52efbbaea3a..116c980b5d29bd2b1c5954143620faf57fb83f9c 100644
--- a/client.d.ts
+++ b/client.d.ts
@@ -82,9 +82,9 @@ declare module '*.gif' {
   const src: string
   export default src
 }
-declare module '*.svg' {
-  const src: string
-  export default src
+declare module '*.svg' { // Using with vite-plugin-svgr
+  const SvgElement: typeof import('*.svg?react')['default']
+  export default SvgElement
 }
 declare module '*.ico' {
   const src: string
@@ -99,6 +99,16 @@ declare module '*.avif' {
   export default src
 }
 
+// cursors
+declare module '*.cur' {
+  const src: string
+  export default src
+}
+declare module '*.ani' {
+  const src: string
+  export default src
+}
+
 // media
 declare module '*.mp4' {
   const src: string
